### DISCUSSION TITLE
How can I disable the login screen in Elsa Studio 3.2.0

### DISCUSSION THREAD

**Discussion by jirimatejka** [elsa-workflows/elsa-studio#286] [Q&A]
I have an Elsa Studio app, where I modified the server authentication to use api key, the key is sent with each request to Elsa server and the server accepts it. I need to disable to Elsa studio login screen... How can I do it?

I found a recommendation to use
builder.Services.AddShell(options => configuration.GetSection("Shell").Bind(options));
and modify appsettings with 
"Shell": {
  "DisableAuthorization": true
}

but it does not work, the login screen is still displayed. 

thanks!

Jiri

**Comment by anupam-mangal** [comment]
Not sure if this helps but the entire security can be disabled in Elsa by adding the below line to the Program.cs :

Elsa.EndpointSecurityOptions.DisableSecurity();

**Reply by MANOJKUMARCH** [reply]
Hi @jirimatejka, we are looking for a similar workaround and want to see if you have got the solution to remove the Login screen?

Your inputs would be of great help for us.

**Reply by jirimatejka** [reply]
Hello, nope I haven't found a way how to do that. It was not critical for us, so I left it with the default login page.

**Comment by sfmskywalker** [comment]
Short answer: in the current Studio codebase, `Shell.DisableAuthorization` only disables the **UI-side** authorization gate. It does not remove the login module or change backend security by itself.

What the current code does:

- `Elsa.Studio.Shell/App.razor` switches from `AuthorizeRouteView` to a plain `RouteView` when `DisableAuthorization` is `true`.
- The login module still registers `UnauthorizedComponentProvider<RedirectToLogin>` and the login-related auth services in `Elsa.Studio.Login/Extensions/ServiceCollectionExtensions.cs`.
- The stock server host still wires the legacy login UI path when `Authentication:Provider` is `ElsaLogin` via `builder.Services.AddLoginModule().UseElsaIdentity()`.

So if you still see the login page, the usual causes are:

1. the `Shell` section is not actually being bound in your custom Studio app,
2. you are still running the `ElsaLogin` module path in the host,
3. or Studio routing is disabled correctly, but the backend still returns `401/403` because server-side auth is separate.

For a login-free Studio setup, I would verify these independently:

- your Studio app really calls `AddShell(options => configuration.GetSection("Shell").Bind(options))`,
- `appsettings.json` actually contains `"Shell": { "DisableAuthorization": true }`,
- you only keep the auth module pieces you still need for your API-key flow,
- and the Elsa backend accepts the API-key-authenticated requests without expecting the built-in login flow.

So the key point is: `DisableAuthorization` is necessary for removing the Studio login screen, but by itself it is not a full replacement for the built-in login/auth stack.

If you want, post the relevant part of your Studio `Program.cs` where `AddShell(...)` and the auth modules are registered, because that is usually where this misconfiguration shows up.
